src/KGzocha/Searcher/Chain/Collection/CellCollection.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace KGzocha\Searcher\Chain\Collection;
+
+use KGzocha\Searcher\AbstractCollection;
+use KGzocha\Searcher\Chain\CellInterface;
+
+/**
+ * Collection of CellInterface objects. It will throw \InvalidArgumentException if tried to use it with less
+ * than MINIMUM_CELLS.
+ *
+ * @author Krzysztof Gzocha <krzysztof@propertyfinder.ae>
+ */
+class CellCollection extends AbstractCollection implements CellCollectionInterface
+{
+    const MINIMUM_CELLS = 2;
+
+    /**
+     * @inheritDoc
+     */
+    protected function isItemValid($item)
+    {
+        return $item instanceof CellInterface;
+    }
+
+    /**
+     * @return \ArrayIterator
+     */
+    public function getIterator()
+    {
+        $this->validateNumberOfCells();
```

Comment:
I would like to have possibility to create empty collection and then make use of `addNamedCell` or `addCell`, but I need to be sure there will be at least `MINIMUM_CELLS` when using this collection